src/main/java/com/toptrumps/core/engine/Game.java

```diff
@@ -116,7 +119,36 @@ public void start() {
                 outcome = new RoundOutcome(DRAW, winners);
                 dealer.putCardsOnCommunalPile(roundCards);
             }
-            listener.onRoundEnd(outcome);
+
+            //Determine if any players have been eliminated
+            Player eliminatedPlayer = null;
+            ArrayList<Player> eliminatedPlayers = new ArrayList<Player>();
+            for(Player player: players){
+                if(player.getDeck().isEmpty()){
+                    if(eliminatedPlayer == null){
+                        eliminatedPlayer = player;
```

Comment:
We can get the eliminated player from just filtering players with empty decks. It's a minor tweak in the current logic, like this:

```
   ArrayList<Player> defeatedPlayers = (ArrayList<Player>) players.stream().filter(p -> p.getDeck().isEmpty()).collect(toList());
            players.removeAll(defeatedPlayers);
```